src/main/java/org/eclipse/basyx/submodel/restapi/vab/VABSubmodelAPI.java

```diff
@@ -97,28 +104,81 @@ public void addSubmodelElement(String idShortPath, ISubmodelElement elem) {
 		getElementProvider().createValue(SubmodelAPIHelper.getSubmodelElementPath(idShortPath), elem);
 	}
 
+	@SuppressWarnings("unchecked")
 	@Override
 	public void deleteSubmodelElement(String idShortPath) {
+		ISubmodelElement submodelElement = getSubmodelElement(idShortPath);
+		if (File.isFile((Map<String, Object>) submodelElement)) {
+			File file = File.createAsFacade((Map<String, Object>) submodelElement);
+			java.io.File tmpFile = new java.io.File(file.getValue());
+			tmpFile.delete();
+		}
 		getElementProvider().deleteValue(SubmodelAPIHelper.getSubmodelElementPath(idShortPath));
```

Comment:
I'm not sure if this is the right point for handling the file Delete. Alternatively, this can be handled by adding a FileSubmodelElementProvider and adding it in SubmodelElementProvider